src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/LambdaBodyLengthCheck.java

```diff
@@ -0,0 +1,234 @@
+ * </p>
+ * <pre>
+ * class Test {
+ *   Runnable r = () -&gt; { // violation, 6 lines
+ *       System.out.println(2); // line 2 of lambda
+ *       System.out.println(3);
+ *       System.out.println(4);
+ *       System.out.println(5);
+ *   };
+ *
+ *   Runnable r2 = () -&gt; // violation, 6 lines
+ *       "someString".concat("1")
+ *                   .concat("2")
+ *                   .concat("3")
+ *                   .concat("4")
+ *                   .concat("5")
+ *                   .concat("6");
+ *
+ *   Runnable r3 = () -&gt; { // ok, 5 lines
+ *       System.out.println(2);
+ *       System.out.println(3);
+ *       System.out.println(4);
+ *   };
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code maxLen.lambdaBody}
+ * </li>
+ * </ul>
+ *
+ * @since 8.37
+ */
+@StatelessCheck
+public class LambdaBodyLengthCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "maxLen.lambdaBody";
+
+    /** Default maximum number of lines. */
+    private static final int DEFAULT_MAX = 10;
+
+    /** Specify the maximum number of lines allowed. */
+    private int max = DEFAULT_MAX;
+
+    /**
+     * Setter to specify the maximum number of lines allowed.
+     *
+     * @param length the maximum length of lambda body.
+     */
+    public void setMax(int length) {
+        max = length;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.LAMBDA};
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (ast.getParent().getType() != TokenTypes.SWITCH_RULE) {
+            final int length = getLength(ast);
+            if (length > max) {
+                log(ast, MSG_KEY, length, max);
+            }
+        }
+    }
+
+    private static int getLength(DetailAST ast) {
+        final DetailAST lambdaBody = ast.getLastChild();
+        final int length;
+        if (lambdaBody.getType() == TokenTypes.SLIST) {
+            length = lambdaBody.getLastChild().getLineNo() - lambdaBody.getLineNo();
+        }
+        else {
+            length = getLastNodeLineNumber(lambdaBody) - getFirstNodeLineNumber(lambdaBody);
+        }
+        return length + 1;
+    }
+
+    /**
+     * Get last node line number.
+     *
+     * @param lambdaBody lambda body node.
+     * @return last node line number.
+     */
+    private static int getLastNodeLineNumber(DetailAST lambdaBody) {
+        DetailAST node = lambdaBody;
+        int result;
+        do {
+            result = node.getLineNo();
+            node = node.getLastChild();
```

Comment:
How sure are we can always use last node? Some nodes like `+` re-arrange the tokens so the last may not always have the max line number.

Reply:
I am not sure how I can prove it, it just seems reasonable.
Can you give examples of specific tokens in expression, where left outmost child can have not lowest line and right outmost can have not biggest?

Reply:
I am not sure what you mean by "left" child, but here is an example tree that applies to one of these comments and not the other. I don't have any examples of the reverse off the top of my head. It may not happen. Note its not a lambda, as I am not very familar with them but it should be possible to create one from it.

````
String s
=
"a"
+
"b"
;

    |      |--VARIABLE_DEF -> VARIABLE_DEF [3:0]
    |      |  |--MODIFIERS -> MODIFIERS [3:0]
    |      |  |--TYPE -> TYPE [3:0]
    |      |  |  `--IDENT -> String [3:0]
    |      |  |--IDENT -> s [3:7]
    |      |  `--ASSIGN -> = [4:0]
    |      |      `--EXPR -> EXPR [6:0]
    |      |          `--PLUS -> + [6:0]
    |      |              |--STRING_LITERAL -> "a" [5:0]
    |      |              `--STRING_LITERAL -> "b" [7:0]
    |      |--SEMI -> ; [8:0]
````

Note "a" and "b" are not in line order compared to `PLUS` token.

Reply:
yes, but it should not be a problem, since code digs down to lowest level and checking outmost left leaf of the tree to outmost right leaf of the tree.
Even if this case order of PLUS token does not matter since code goes down to STRING_LITERAL tokens, which are ordered correctly. Also, there is a test case for plus expression.

Reply:
Also there is same way to get lambda last child in other check and there is no issues (code was added more than year ago)
https://github.com/checkstyle/checkstyle/blob/de2f537d42a0165709c40b74a1757822814d22eb/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java#L485-L491